pkg/scheduler/queue/queue.go

```diff
@@ -337,7 +338,9 @@ func (q *RequestQueue) dispatcherLoop() {
 			for currentElement != nil {
 				waitingDequeueReq := currentElement.Value.(*QuerierWorkerDequeueRequest)
 				waitingDequeueReq.sendError(ErrStopped)
-				currentElement = currentElement.Next()
+				nextElement := currentElement.Next()
+				q.waitingDequeueRequestsToDispatch.Remove(currentElement)
+				currentElement = nextElement
```

Comment:
@chencs So previously we didn't drop these elements from the queue after sending the "scheduler stopping error" to their receiver channel.
It did not affect anything, but dropping them makes it possible to assert on the desired outcome in our tests.